prog/vm/vm_pool.rb

```diff
@@ -61,7 +61,7 @@ def before_run
 
   label def wait
     if vm_pool.size - vm_pool.vms.count > 0
-      idle_cpus = VmHost.where(allocation_state: "accepting", arch: vm_pool.arch, location: ["github-runners", "hetzner-hel1", "hetzner-fsn1"]).select_map { sum((total_cores - used_cores) * total_cpus / total_cores) }.first.to_i
+      idle_cpus = VmHost.where(allocation_state: "accepting", arch: vm_pool.arch, location_id: Location.where(name: ["github-runners", "hetzner-hel1", "hetzner-fsn1"]).select_map(:id)).select_map { sum((total_cores - used_cores) * total_cpus / total_cores) }.first.to_i
```

Comment:
```suggestion
      idle_cpus = VmHost.where(allocation_state: "accepting", arch: vm_pool.arch, location_id: Location.where(name: ["github-runners", "hetzner-hel1", "hetzner-fsn1"]).select(:id)).select_map { sum((total_cores - used_cores) * total_cpus / total_cores) }.first.to_i
```

Better to use a subquery here than a separate query to get 3 ids.  It would be even better to use three constant references and avoid the need for a separate query or a subquery.